### DISCUSSION TITLE
Imaginary not producing any logs even with `-log-level debug`

### DISCUSSION THREAD

**Discussion by llucax** [nextcloud/all-in-one#2225] [Questions] [answered]
Hi, I'm using imaginary from the nextcloud-aio docker image and it seems to be working fine. When new images are viewed, I see the imaginary process taking CPU and images are popping up correctly, but I don't get any logs, even when using `-log-level debug`.

I configured it as a systemd services using this unit:
```systemd
[Unit]
Description=Imaginary image processing service
After=docker.service
Requires=docker.service

[Service]
TimeoutStartSec=0
Restart=always
ExecStartPre=-/usr/bin/docker stop %n
ExecStartPre=-/usr/bin/docker rm %n
ExecStartPre=/usr/bin/docker pull nextcloud/aio-imaginary
ExecStart=/usr/bin/docker run \
        --rm \
        --name %n \
        --cap-add=sys_nice \
        -p 127.0.0.1:9000:9000 \
        nextcloud/aio-imaginary \
                -return-size \
                -concurrency 20 \
                -enable-url-source \
                -log-level debug
ExecStop=/usr/bin/docker stop %n

[Install]
WantedBy=multi-user.target
```

I also tried running the binary manually using `docker` and can't get any logs.

Since stuff was working, I sort of forget about it, but now I'm having some issues with the `facerecognition` app, which is saying:

```
Faces found: 0. Image will be skipped because of the following error: Imaginary image response is not valid.
```

So it would be nice to be able to have some idea of what imaginary is doing. I notice today that I got one single message in the logs from imaginary (or I guess it is):
```
Mar 28 18:37:35 nube docker[326218]: (process:1): VIPS-WARNING **: 16:37:33.905: VipsJpeg: large XMP not saved
```

Anyway. Any ideas why I'm not getting the regular HTTP server logs when using `-log-level debug` and how can I enable them?
I saw https://github.com/nextcloud/all-in-one/discussions/2013 but in my case I would think imaginary is really working because I can see it busy in `top` when I go look at a folder that didn't have previews generated before.

Thanks!

**Comment by szaimen** [comment]
Hi, I fear I cannot help here much since this should rather be asked at https://github.com/h2non/imaginary but it looks strange to me that you use systemd to start/restart the container instead of just using e.g. `--restart always` in the docker run command directly...

**Reply by llucax** [reply]
Well, the image is provided by this project, that's why I thought I could ask first here to see if I'm the only one experiencing this issue. On a google search I found nobody else reporting logging not being working because maybe nobody checks the logs, or maybe imaginary just doesn't produce any logs.

I don't see anything strange in running a service via systemd, it's the standard way to start and monitor any running service in my server, so I don't see any reason to go with something different. But again, it doesn't produce logs by running `docker run` manually either.

**Reply by llucax** [reply]
BTW, the `-p` option is also not changing the bind port, there seems to be something really funky with command-line parsing, I'm surprised this is not reported more often.

```
nobody@8349e8a27299:/$ /usr/local/bin//imaginary -p 901 -concurrency 10 -log-level debug -enable-url-source
2023/03/29 20:40:00 listen: listen tcp :9000: bind: address already in use
```

I checked with the `h2non/imaginary` docker image and it is the same, but that image doesn't implement `-return-size` which is recommended in the nextcloud docs (it also recommends the aio container saying that the official is too old, which it is).

I really wonder if this project is completely abandoned (although there are commits in Jan, which is not that bad).

**Comment by szaimen** [comment]
You would likely meed to overwrite the entrypoint to make your options work but I fear this is usual docker stuff and pretty much unrelated to AIO. 

See https://docs.docker.com/engine/reference/run/#entrypoint-default-command-to-execute-at-runtime

**Reply by llucax** [reply]
I don't see how overriding the entrypoint could help really, this fails even using `docker exec h2non/imaginary /bin/bash` and then running the `imaginary` binary manually, as shown in the example about `-p` being ignored too.

**Reply by llucax** [reply]
Anyway, I don't want to distract you any further, I wrote this as a discussion to just see if there were other users having similar issues or tips. I'm not saying this is a bug in this project nor that you should fix anything. I think it is related to AIO though, as this image is a component of it.

**Reply by llucax** [reply]
One more curious thing is to set up imaginary I followed this nice guide:

https://okxo.de/speed-up-nextcloud-preview-generation-with-imaginary/

In there, when they tell how to test is working, they show their imaginary producing logs as I would have expected, and they are using the aio image, this is what also puzzled me. It seems like it have worked at some point.

**Comment by szaimen** [comment]
Maybe you should use docker-compose then instead of systemd? That would make much more sense imho and is documented to work: https://okxo.de/speed-up-nextcloud-preview-generation-with-imaginary/

**Reply by llucax** [reply]
Thanks for the suggestion but I don't think that would help in any way. As I said I tried running the binary manually inside the docker container, I can't see how docker compose can really help here.

**Comment by szaimen** [comment, marked as answer]
I guess it is best then to indeed ask your question at https://github.com/h2non/imaginary as we only package their Docker image and source code here.

**Comment by otmi100** [comment]
I had the same problem. Changing the log level to **info** solved it for me.
